### ISSUE TITLE
TimeGrad: Adding Covariates as Conditioning Input

### ISSUE THREAD

**Issue by ClaraGrthns** [zalandoresearch/pytorch-ts#169]
Hi there, 

I was wondering whether it is possible to add covariates into the conditioning part of the TimeGrad implementation? It seems to me as if the transformations that are done automatically inside the time_grad_estimator for time features and lagged sequences are used as conditioning input, but how could I incorporate further covariates (i.e. of the type FieldName.feat_dynamic_real), that I don't want to forecast.

Thanks in advance!

**Comment by kashif** [comment]
so @ClaraGrthns yes the model does take in date-time features and lags as covariates, to add further covariates one can have a look at how that is done for the multivariate setting in the deepAR model in the 0.7.0 branch... 

i didn't see the need to add further covaraites in the multivariate setting... even lags are too much as each lag index causes the multivariate dim to go up by the multivariate dim which can be high especially for datasets of multivariate size 2K or more...

also there was no need to add an id for each variate as the order of the variate is kept fixed... 

but yeah if you have more covariates for your data, then just concat them to the input vectors as in the deepAR setting...

**Comment by zysongg** [comment]
> so @ClaraGrthns yes the model does take in date-time features and lags as covariates, to add further covariates one can have a look at how that is done for the multivariate setting in the deepAR model in the 0.7.0 branch...
> 
> i didn't see the need to add further covaraites in the multivariate setting... even lags are too much as each lag index causes the multivariate dim to go up by the multivariate dim which can be high especially for datasets of multivariate size 2K or more...
> 
> also there was no need to add an id for each variate as the order of the variate is kept fixed...
> 
> but yeah if you have more covariates for your data, then just concat them to the input vectors as in the deepAR setting...

hello，The predictor outputs all nans in the TimeGrad. Could you slove it?  thanks.

**Comment by SimonTreu** [comment]
@ClaraGrthns did you adapt the TimeGrad implementation to add covariates? I would be interested in that as well.

**Comment by ClaraGrthns** [comment]
@SimonTreu no, in the end, I chose another Time Series Diffusion Model to incorporate future covariates (weather forceasts) into the conditioning input of the denoising network. It definitely improved the point estimates on the target time series, but at the same time it worsened the sample diversity. Have you tried something similar?
